### ISSUE TITLE
[13] Tracback on Adding File/Image without Selecting any File

### ISSUE THREAD

**Issue by sswapnesh** [odoo/odoo#42448]
Impacted versions: 13.0

Steps to reproduce:
![Screenshot (35)](https://user-images.githubusercontent.com/16624719/71560549-687f0200-2a91-11ea-8522-f7fd24cc1323.png)

![Screenshot (36)](https://user-images.githubusercontent.com/16624719/71560551-77fe4b00-2a91-11ea-8f8f-638bc230d112.png)

Current behavior:
```
Traceback:
WrappedRange/this.insertNode@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:155:313
onSave@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/editor/rte.summernote.js:494:15
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:224:31
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:328:42
init/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/wysiwyg/widgets/dialog.js:44:18
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:224:31
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:328:42
destroy@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:232:18
OdooClass.extend/</prototype[name]</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/class.js:90:38
close@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:214:14
save@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/wysiwyg/widgets/dialog.js:66:14
save/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/wysiwyg/widgets/media_dialog.js:164:20
promise callback*save@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/wysiwyg/widgets/media_dialog.js:161:41
OdooClass.extend/</prototype[name]</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/class.js:90:38
_setButtonsTo/</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:303:44
dispatch@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5183:27
add/elemData.handle@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4991:28
EventListener.handleEvent*add@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5039:12
on/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4939:16
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:354:19
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:189:17
on@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4938:14
on@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5547:10
_setButtonsTo/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:300:21
_.forEach@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/underscore/underscore.js:145:17
_setButtonsTo@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:291:11
set_buttons@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:161:14
willStart/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:136:22
promise callback*willStart@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:113:51
OdooClass.extend/</prototype[name]</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/class.js:90:38
_widgetRenderAndInsert@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/widget.js:427:21
appendTo@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/widget.js:193:21
open@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/dialog.js:193:14
open@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/tools/debug_manager_backend.js:760:32
OdooClass.extend/Class.include/</prototype[name]</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/class.js:123:38
open@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/wysiwyg/widgets/dialog.js:75:28
OdooClass.extend/</prototype[name]</<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/class.js:90:38
_onMediaDialogDemand@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/editor/rte.summernote.js:1287:21
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:224:31
trigger@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/src/js/core/mixins.js:328:42
eventHandler.modules.imageDialog.showImageDialog@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/editor/rte.summernote.js:484:14
ImageDialog/this.show@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:291:475
showImageDialog@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:381:994
hToolbarAndPopoverClick@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:388:346
dispatch@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5183:27
add/elemData.handle@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4991:28
EventListener.handleEvent*add@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5039:12
on/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4939:16
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:354:19
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:189:17
on@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:4938:14
on@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:5547:10
EventHandler/this.attach@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:400:450
eventHandler.attach@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:680:221
eventHandler.attach@http://832599-13-0-caa4c7.runbot29.odoo.com/web_editor/static/src/js/editor/rte.summernote.js:887:15
summernote/<@http://832599-13-0-caa4c7.runbot29.odoo.com/web/content/1574-7600c60/web_editor.summernote.js:424:235
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:354:19
each@http://832599-13-0-caa4c7.runbot29.odoo.com/web/static/lib/jquery/jquery.js:189:17
```

Expected behavior: No traceback / Close Wizard

Video/Screenshot link (optional):

**Comment by mart-e** [comment]
@sswapnesh could you please give your tracebacks in debug=assets mode  and also copy the content (instead of giving a screenshot). Thanks !

@Yenthe666 why website?

@dmo-odoo looks like `media` can be undefined when trying to inject it.

https://github.com/odoo/odoo/blob/caa4c7ef6d23f695827b58e60a013a9438eb67cf/addons/web_editor/static/src/js/editor/rte.summernote.js#L494

**Comment by Yenthe666** [comment]
Not so well labeling, mea culpa!

**Comment by pedrobaeza** [comment]
Any news on this issue?

**Comment by sswapnesh** [comment]
I can reproduce the same. 

![screenshot-localhost_8069-2020 04 24-23_32_58](https://user-images.githubusercontent.com/16624719/80243002-f9529c80-8683-11ea-9c6b-d91b9aeed85f.png)

```
Error:


Traceback:
TypeError: Failed to execute 'insertBefore' on 'Node': parameter 1 is not of type 'Node'.
    at WrappedRange.insertNode (http://localhost:8069/web/content/436-107606f/1/web_editor.summernote.js:155:313)
    at Class.onSave (http://localhost:8069/web_editor/static/src/js/editor/rte.summernote.js:503:15)
    at Class.trigger (http://localhost:8069/web/static/src/js/core/mixins.js:224:31)
    at Class.trigger (http://localhost:8069/web/static/src/js/core/mixins.js:328:42)
    at Class.<anonymous> (http://localhost:8069/web_editor/static/src/js/wysiwyg/widgets/dialog.js:45:18)
    at Class.trigger (http://localhost:8069/web/static/src/js/core/mixins.js:224:31)
    at Class.trigger (http://localhost:8069/web/static/src/js/core/mixins.js:328:42)
    at Class.destroy (http://localhost:8069/web/static/src/js/core/dialog.js:232:18)
    at Class.prototype.(anonymous function) [as destroy] (http://localhost:8069/web/static/src/js/core/class.js:90:38)
    at Class.close (http://localhost:8069/web/static/src/js/core/dialog.js:214:14)
```

**Comment by mart-e** [comment]
There is a fix at #46280 waiting for a review ;-)

**Comment by Yenthe666** [comment]
Fixed in V11 at https://github.com/odoo/odoo/commit/9baf753ace725b000eae98cd95e4f5e0aa84466a and will very soon be forward ported into 12/13.
